### PR TITLE
Increase Code Coverage for 'javaparser/ast/body' package

### LINKED ISSUE #820: Increase code coverage
We have currently a code coverage of ~60%. We could look into increasing it, perhaps focusing on low hanging fruits. I do not think it makes sense to aim at a very high value but I suppose that with a limited effort we could move the needle.

### PR DESCRIPTION
**Pull request by YinyinChencr7**
Fixes #820 

This PR aims to address the "Increase code coverage" issue by adding additional tests for 'javaparser/ast/body' package.

- Added comprehensive unit tests for the 'AnnotationMemberDeclaration' class and other classes in the 'javaparser/ast/body' package, enhancing test coverage for the 'javaparser/ast/body' package.
- Tested various annotation methods, type casting, and method declaration functionalities.
- Improved code quality and reliability by ensuring various edge cases are handled correctly through test cases.

Code coverage for 'javaparser/ast/body' package before: 54.57%

image

Code coverage for 'javaparser/ast/body' package after: 72.53%

image

### CODE REVIEW

**Comment by jlerbsc** [comment]
Thank you for this work but could you suggest PRs which are more targeted and which impact much fewer source files. I will no longer validate this type of PR.

File: pom.xml [hunk 1]
```diff
@@ -329,7 +329,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-dependency-plugin</artifactId>
-                    <version>3.6.1</version>
+                    <version>3.6.0</version>
```
Comment (jlerbsc) [inline pom.xml]: Why did you modify this version?

File: pom.xml [hunk 2]
```diff
@@ -304,7 +304,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>3.2.1</version>
+                    <version>3.1.2</version>
```
Comment (jlerbsc) [inline pom.xml]: Why did you modify this version?

File: javaparser-core-testing/pom.xml [hunk 3]
```diff
@@ -103,9 +103,8 @@
     <dependencies>
         <dependency>
             <groupId>com.github.javaparser</groupId>
-            <artifactId>javaparser-core</artifactId>
-            <version>${project.version}</version>
-            <scope>test</scope>
+            <artifactId>javaparser-symbol-solver-core</artifactId>
+            <version>3.25.5</version>
```
Comment (jlerbsc) [inline javaparser-core-testing/pom.xml]: I don't understand this commit?

**Comment by jlerbsc** [comment]
There are modifications in this PR that I don't understand on pom files. This is why I am going to close the PR. I suggest you resubmit your work in several successive PRs containing mutually consistent modifications.

File: javaparser-core-testing/pom.xml (hunk 3 above)
Comment (YinyinChencr7) [reply javaparser-core-testing/pom.xml]: Sorry for my oversight. I mistakenly pushed changes to the pom files into this branch. I understand the concerns raised and will correct these errors promptly. I'll then split my contributions and open a new PR as you've suggested. Thank you for pointing out the inconsistencies.
